src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs

```diff
@@ -320,6 +325,23 @@ private void WriteMvidsTo(Metadata metadata, ObjectWriter writer, CancellationTo
             WriteMvidTo((ModuleMetadata)metadata, writer, cancellationToken);
         }
 
+        private static bool TryGetModules(AssemblyMetadata assemblyMetadata, out ImmutableArray<ModuleMetadata> modules)
```

Comment:
Could we make this an extension method and share it with code in  #17687?

Reply:
Yes, let me make this an extension.

Reply:
Actually, seems that callsite has a try-catch for two different method calls and also returns modules - using this extension would just make the code in there more icky. I'll stick with the current code style for now..